cameraconnect/src/cameras.py

```diff
@@ -405,22 +409,47 @@ def _connect(self) -> None:
         # If multiple cameras in device list, choose the right
         #   one by changing the list_index or by using another
         #   argument
-
+        first = True  # in case one camera is detected multiple times
+        self.__remove_duplicate_entry_from_harvester()
         for camera in self.h.device_info_list:
-            #read cameras mac address
-            #ATTENTION: only works with BAUMER SDK
-            device_mac_address = str(camera.id_).replace("_","").replace("devicemodul","")
-            if device_mac_address.upper() == self.mac_address.upper().replace(":",""):
+            # read cameras mac address
+            # ATTENTION: only works with BAUMER SDK
+            device_mac_address = str(camera.id_).replace("_", "").replace("devicemodul", "")
+            logging.debug(
+                f"current device_mac_address: {device_mac_address.upper()}, {self.mac_address.upper().replace(':', '')}")
+
+            if not first:
+                logging.debug(f"camera {camera} is not first one with target  id")
+                continue
+            if device_mac_address.upper() == self.mac_address.upper().replace(":", ""): # fixme id detection system is in need of improvement, fix plz
+
                 try:
+                    logging.debug(f"attempting to connect to device {camera.id_}")
                     self.ia = self.h.create_image_acquirer(id_=camera.id_)
-                except:
-                    logging.error("Camera ist not reachable. Most likely another container already occupies the same camera. One camera can only be used by exactly one container at the time.")
+                    first = False
+                except Exception as _e:
+                    logging.error(
+                        "Camera is not reachable. Most likely another container already occupies the same camera. "
+                        f"One camera can only be used by exactly one container at any time. {_e}")
                     sys.exit("Camera not reachable.")
                 logging.debug("Using:" + str(camera))
                 logging.debug(HORIZONTAL_CONSOLE_LINE)
+            else:
```

Comment:
debug only full rework in progress

Reply:
requires checks for different camera vendors cti files, since the id_  property from harvesters is not necessarily the mac address but sometimes something else e.g. mac adress with : between every double octet and ip attached or mac address with out : but device type in the beginning.